README.md

```diff
@@ -1,20 +1,253 @@
+  [{"transactionID":"6691791616826663170","grandTotal":45.59, ... }]
+  [2] insert output
+  ["Order Completed",{"transactionID":"6691791616826663170","grandTotal":45.59, ... }]
+  [3] suffix output
+  ["Order Completed",{"transactionID_str":"6691791616826663170","grandTotal_real":45.59, ... }]
+  [4] function output
+  [null]
+dataLayer.transaction[!(items)] handleData exit
+[null]
+
+Order Completed {transactionID_str: "6691791616826663170", grandTotal_real: 45.59, ... }
+```
+
+The data emitted from a data layer can be seen below the `handleData entry` line.  Each operator will print its respective output.  The order of operators is denoted by `[n]` just before the operator's name.  Eventually `[null]` is printed because this is the result of the function call.  The `Order Completed` line with the corresponding object is the console log showing the arguments that were sent to the function call.
+
+In addition to showing the data conversions, debug includes the following statistics.
+
+- The number of properties in the object shown as `numKeys`.
+- The size (assuming UTF16) of the values in the object shown as `sizeOfValues`.
+- The size (assuming UTF16) of the object when it is stringified shown as `sizeOfPayload`.
+
+## Deploying Rules
+
+Once rules have been verified in a test environment, they can be moved to production.  New rules can be added to any existing rules such as those in `window['_dlo_rules']`.  Alternatively, new rules can be placed in their own rule set to allow contribution from different teams or parts of a site.  This can be done by creating a new configuration option that begins with `_dlo_rules` - for example `window['_dlo_rules_cart_and_checkout']`.  When DLO initializes all properties that begin with `_dlo_rules` will be processed.
+
+After new rules are deployed, visit the site and ensure the data is being sent to the destination.  One approach is to simply review in FullStory that the custom event or user variable has been set.  If data is not found, set the `debug` property to `true` for any problematic rules.  If problems still persist, see the Monitoring section to verify something else isn’t at fault.
+
+## Monitoring
+
+By default, DLO will output any logging related messages to the browser’s JavaScript console.  For production environments, a custom log appender is preferable, and it will allow you to redirect logging to a centralized location.
+
+Below is an example of adding a custom FullStory log appender. (Be aware of [rate limiting](https://help.fullstory.com/hc/en-us/articles/360020623234#custom-property-rate-limiting) if sending logs to FullStory; however, DLO only logs when errors occur.)  When a log event is generated by DLO, it will pass the event to the custom appender, which sends the information to FullStory as a custom event.
+
+```javascript
+window['_dlo_appender'] = {
+ log: ({ level: level_int, message: message_str, datalayer: datalayer_str }) => {
```

Comment:
It might be a good idea to describe what you're doing in this code sample a little more. Something like "we're using JavaScript [Aliases](https://davidwalsh.name/destructuring-alias) to add the [type suffixes](https://help.fullstory.com/hc/en-us/articles/360020623234-FS-Recording-Client-API-Requirements#custom-property-name-requirements) expected by the FullStory API.

Reply:
I think this section probably needs to just fold into a "batteries included" snippet we put in the QuickStart.  And then in Monitoring we can make not of how to do that.